tests/test_dag.py

```diff
@@ -167,6 +194,51 @@ def test_multi_dependencies_dag(self):
         self.assertEqual(node_5.result(), 8)
         self.assertEqual(node_6.result(), 24)
 
+        # The order of this assertion is somewhat fragile. It is deterministic
```

Comment:
great comment! will be super-helpful if/when it fails in precisely that way & someone with less context comes along to fix it. <3